Open Judge System/Web/OJS.Web/Controllers/TempController.cs

```diff
@@ -172,16 +173,32 @@ public ActionResult MigrateProblemsToSharedGroups()
 
         public ActionResult MigrateContestStartEndTimeOfParticipantsInParticipationStartEndTime()
         {
-            var affected = this.Data.Participants
-                .All()
-                .Where(p => p.ContestStartTime != null || p.ContestEndTime != null)
-                .Update(p => new Participant
-                {
-                    ParticipationStartTime = p.ContestStartTime,
-                    ParticipationEndTime = p.ContestEndTime,
-                    ContestStartTime = null,
-                    ContestEndTime = null
-                });
+            var affected = 0;
+
+            using (var scope = TransactionsHelper.CreateTransactionScope())
+            {
+                affected += this.Data.Participants
+                    .All()
+                    .Where(p => p.IsOfficial && (p.ContestStartTime != null || p.ContestEndTime != null))
```

Comment:
Shouldn't the check be .Where(p => p.IsOfficial && p.ContestStartTime != null **&&** p.ContestEndTime != null)? Is there a online constest official participant who has only a ContestStartTime or an ConetstEndTime?

Reply:
Currently there is not, but I want to be sure that all the values are properly copied from one column to the other, without any risk of data loss, and then if there are any anomalies I will be able to see them.